analyzer/queries/queries.go

```diff
@@ -715,13 +715,19 @@ var (
 
 	RuntimeEVMNFTUpsert = `
     INSERT INTO chain.evm_nfts AS old
-      (runtime, token_address, nft_id, owner, num_transfers, last_want_download_round)
+      (runtime, token_address, nft_id, num_transfers, last_want_download_round)
     VALUES
-      ($1, $2, $3, $4, $5, $6)
-    ON CONFLICT (runtime, token_address, nft_id) DO UPDATE
-    SET
-      owner = COALESCE(excluded.owner, old.owner),
-      num_transfers = old.num_transfers + excluded.num_transfers`
+      ($1, $2, $3, 0, $4)
+    ON CONFLICT (runtime, token_address, nft_id) DO NOTHING`
+
+	RuntimeEVMNFTUpdateTransfer = `
+    UPDATE chain.evm_nfts SET
+      owner = $4,
+      num_transfers = num_transfers + $5
+    WHERE
+      runtime = $1 AND
+      token_address = $2 AND
+      nft_id = $3`
```

Comment:
I would prefer to keep the single-statement setup here. It will keep the upsert a lot more similar to other upserts, both on the Go and SQL side, and thus easier to read and reason about.

We'll just need a special value for `owner` (or rather `$4`) that SQL can recognize. So in the old query, only a single line would need to change, something like
```sql
ON CONFLICT (runtime, token_address, nft_id) DO UPDATE
    SET
      owner = CASE WHEN excluded.owner = '__BURN__' THEN NULL ELSE COALESCE(excluded.owner, old.owner) END,
```

Reply:
"please do this completely differently. approved"

Reply:
Sorry if it looked weird. As explained on slack, approved to unblock you after I'm gone in my timezone.
Also, "completely differently" is ... one possible interpretation in that the whole PR is not very large, so the percentage of affected lines is large. But I agree with all the debugging work that led to the small PR, and with the semantics of the fix, which is to sometimes force the NULL into the DB. Those are the more important parts IMO, and my comment is about a technicality and code readability.
Really signing off now.